lib/src/data_table_2.dart

```diff
@@ -362,23 +377,34 @@ class DataTable2 extends DataTable {
         child: DropdownButtonHideUnderline(child: label),
       ),
     );
-    if (onTap != null) {
+    if (onTap != null ||
+        onDoubleTap != null ||
+        onLongPress != null ||
+        onTapDown != null ||
+        onTapCancel != null) {
       label = InkWell(
         onTap: onTap,
-        child: label,
+        onDoubleTap: onDoubleTap,
+        onLongPress: onLongPress,
+        onTapDown: onTapDown,
+        onTapCancel: onTapCancel,
         overlayColor: overlayColor,
+        child: label,
       );
     } else {
       label = GestureDetector(
         child: TableRowInkWell(
-            child: label,
-            overlayColor: overlayColor,
-            onTap: onRowTap == null
-                ? onSelectChanged
-                : () {
-                    onRowTap();
-                    onSelectChanged();
-                  }),
+          child: label,
+          overlayColor: overlayColor,
+          onTap: onRowTap == null
+              ? onSelectChanged
+              : () {
+                  onRowTap();
+                  onSelectChanged();
+                },
+          onDoubleTap: onRowDoubleTap,
+          onLongPress: onRowLongPress,
```

Comment:
Flutter's DataTable has these two method provided. They work.

But flutter did not provide `onTapDown` and `onTapCancel` here. While we can hook up both outside of `TableRowInkWell`, in the `GestureDetector` along with `onSecondaryTap`, the behaviour is different than having all in the same `InkWell`.

I made a decision to not supporting `onTapDown` and `onTapCancel` because of this reason. Would like to hear what @maxim-saplin thinks.

Reply:
Makes sense, don't see any reason to pursue all tap events until someone asks deliberately